### PR TITLE
KAFKA-4856: make unregisterAppInfo method to synchronized

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
minor fix to avoid error logs
cc @ijuma

File: clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public static void registerAppInfo(String prefix, String id) {
         }
     }
 
-    public static void unregisterAppInfo(String prefix, String id) {
+    public static synchronized void unregisterAppInfo(String prefix, String id) {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java:64]: If this must be `synchronized`, don't we have to do the same for `registerAppInfo`?

Comment (omkreddy) [reply clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java:64]: yes for consistency, we can synchronize registerAppInfo method also. when multiple clients tries register mbean with the same clientId, we print an error message and continue.  when multiple threads try to unregister we delete only if mbean exist in the mbean registry.

updated the PR.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6888/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6874/
Test PASSed (JDK 8 and Scala 2.12).

File: clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java (hunk 1 above)
Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java:64]: Btw, we should take the chance and make `version` and `commitId` final. Something like:

```java
    private static final String version;
    private static final String commitId;

    static {
        Properties props = new Properties();
        try (InputStream resourceStream = AppInfoParser.class.getResourceAsStream("/kafka/kafka-version.properties")) {
            props.load(resourceStream);
        } catch (Exception e) {
            log.warn("Error while loading kafka-version.properties :" + e.getMessage());
        }
        version = props.getProperty("version", "unknown").trim();
        commitId = props.getProperty("commitId", "unknown").trim();
    }
```

Comment (omkreddy) [reply clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java:64]: Thanks. Updated the changes.

**Review by ijuma — APPROVED** [review]
LGTM. I will merge to trunk and 0.11.0 after the tests pass.

**Comment by omkreddy** [comment]
retest this please

**Pull request closed without merging**